src/main/drivers/rtc.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+// Don't access the fields directly. Use the getter
+// functions below.
+typedef struct {
+    int32_t unix;
+    uint32_t nanos;
```

Comment:
Note, that nanoseconds won't fit in a `uint32_t`.

Note also the iNav works in an extremely constrained environment - we have tight constraints on ROM size, RAM size and processor cycles. The easiest way to save these things is not to add unrequired functionality.

There is no current use case for nanosecond time. Especially for the RTC. Internal events are times using the system clock and will continue to be done so.

All nanosecond times should be removed. If we find we need them later we can add them - it should be straightforward to do this without breaking anything.

Reply:
`unix` is a bit of a confusing variable name. `unixTime` is unambiguous.

Reply:
I'd remove the `timestamp_t` struct altogether and do a `typedef unit32_t rtcTimeStamp_t;`

Reply:
@martinbudden nanoseconds fit in 30 bits unsigned, 31 with sign. log2(1e+9) is 29 something. The reason I went with nanoseconds is because using milliseconds will require 16 bits for them, so nanos only needs 16 more bits. Since the GPS reports nanosecond, 16 bits seem like small enough to not make a difference in size/ram usage while making the API future proof.

Reply:
YAGNI = You Ain't Going to Need It. 
What's the use case for nanosecond RTC? For measuring accurate intervals we use `micros()` or system ticks. For date/time - 1 second resolution is more than enough.